FBReader/src/main/java/org/geometerplus/android/fbreader/library/DownloadedBookInfoActivity.java

```diff
@@ -109,11 +110,17 @@ protected void onCreate(Bundle icicle) {
 	}
 
 	@Override
-	protected void onStart() {
-		super.onStart();
+	protected boolean shouldShowAddToReadingListButton(){
+		if(isBookshareIdAvailable) return super.shouldShowAddToReadingListButton();
+		else return false;
+	}
 
-		final Book book = Book.getByFile(myFile);
 
+	@Override
+	protected void onStart() {
+		final Book book = Book.getByFile(myFile);
```

Comment:
It wont, getBookByFile looks up the book by searching for the file, and the code for that already handles the edge cases